### PR TITLE
Update docs

### LINKED ISSUE #9: Document how to use test wrappers
Once we have basic operational we should add docs describing how to use the capabilities with simple examples pulled from files in the repo. Likely a later task relative to basic functionality.

### LINKED ISSUE #17: Share kernel context across multiple tests
We need an option to share kernel context across multiple test functions so that the set up (for a test to be executed) need not be executed multiple times, but just once - which results in faster tests. 

**UPDATE:** Check https://github.com/nteract/testbook/issues/17#issuecomment-646234434.

### LINKED ISSUE #37: Docstrings and type hinting

### PR DESCRIPTION
**Pull request by rohitsanj**
Resolves #37 
Resolves #17 
Resolves #9

### CODE REVIEW

File: docs/getting-started/index.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+# Installation and Getting Started
+
+`testbook` is a unit testing framework for testing code in Jupyter Notebooks.
+
+## Installing `testbook`
+
+<!-- TODO: add package name -->
+
+```{code-block} bash
+pip install ...
```
Comment (MSeal) [inline docs/getting-started/index.md:22]: Can change this to `nteract-testbook`

File: testbook/client.py [hunk 2]
```diff
@@ -1,32 +1,37 @@
...
+        Return a reference to an object in the kernel
+        """
+
         # Check if exists
         self.inject(name)
         return TestbookObjectReference(self, name)
 
     @staticmethod
-    def _construct_call_code(func_name, args=None, kwargs=None):
+    def _construct_call_code(func_name: str, args=None, kwargs=None) -> str:
```
Comment (MSeal) [inline testbook/client.py:34]: I think you can also type args as `Optional[List]` and kwargs as `Optional[Dict]`

File: tox.ini [hunk 3]
```diff
@@ -61,6 +61,6 @@ basepython =
     flake8: python3.8
     manifest: python3.8
     dist: python3.8
-    docs: python3.8
+    docs: python3.7
```
Comment (MSeal) [inline tox.ini:64]: Had to drop for nbsphinx?

File: docs/usage/index.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+```{code-block} python
+@testbook.testbook('/path/to/notebook.ipynb', execute=True)
+def test_say_hello(tb):
+    Foo = tb.ref("Foo")
+    bar = Foo("bar")
+
+    assert bar.say_hello() == "Hello bar!"
+```
+
+## Share kernel context across multiple tests
```
Comment (rohitsanj) [inline docs/usage/index.md:116]: @MSeal

**Pull request merged**